internal/encoded.go

```diff
@@ -237,36 +269,270 @@ func getEncoding(payload *commonpb.Payload) (string, error) {
 	return "", ErrEncodingIsNotSet
 }
 
-func toString(payload *commonpb.Payload) (string, error) {
-	result := ""
+type PayloadConverter interface {
+	// ToPayload single value to payload.
+	ToPayload(value interface{}) (*commonpb.Payload, error)
+	// FromPayload single value from payload.
+	FromPayload(payload *commonpb.Payload, valuePtr interface{}) error
```

Comment:
Document that it should return an error if valuePtr is of a type it doesn't know how to deal with.